Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/TextTests/DeleteTextTest.cs

```diff
@@ -0,0 +1,55 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore.Query;
+using Moq;
+using Streetcode.BLL.DTO.Streetcode.TextContent;
+using Streetcode.BLL.MediatR.Streetcode.Text.Delete;
+using Streetcode.BLL.MediatR.Streetcode.Text.GetById;
+using Streetcode.DAL.Entities.Streetcode.TextContent;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Streetcode.XUnitTest.StreetcodeTest.TextTest
+{
+    public class DeleteTextTest
+    {
+        private Mock<IRepositoryWrapper> repository;
+
+        public DeleteTextTest()
+        {
+            repository = new Mock<IRepositoryWrapper>();
+        }
+
+        private static Text GetText(int id)
+        {
+            return new Text { Id = id };
+        }
+
+        private static Text? GetTextWithNotExitingId()
+        {
+            return null;
+        }
+
+        [Theory]
+        [InlineData(1)]
+        public async Task DeleteTextSuccessfull(int id)
+        {
+            repository.Setup(x => x.TextRepository.GetFirstOrDefaultAsync(It.IsAny<Expression<Func<Text, bool>>>(),
+                It.IsAny<Func<IQueryable<Text>, IIncludableQueryable<Text, object>>>()))
+                .ReturnsAsync(GetText(id));
+
+            repository.Setup(x => x.TextRepository.Delete(GetText(id)));
+
+            repository.Setup(x => x.SaveChangesAsync()).ReturnsAsync(1);
+
+            var handler = new DeleteTextHandler(repository.Object);
+
+            var result = await handler.Handle(new DeleteTextCommand(id), CancellationToken.None);
+
+            Assert.Multiple(
+            () => Assert.NotNull(result),
+            () => Assert.True(result.IsSuccess));
```

Comment:
As for me these checks are useless. If result is not null (first assert) that the result is obviously successful. Maybe it is better to check the type of result.Value. You can also check if the element was really deleted